packages/@dcl/sdk-commands/src/commands/deploy/index.ts

```diff
@@ -62,31 +63,38 @@ export function help() {
 }
 
 export async function main(options: Options) {
-  const dir = resolve(process.cwd(), options.args['--dir'] || '.')
+  const projectRoot = resolve(process.cwd(), options.args['--dir'] || '.')
   const openBrowser = !options.args['--no-browser']
   const skipBuild = options.args['--skip-build']
   const linkerPort = options.args['--port']
   const { error } = options.components.logger
 
-  const comps = { components: options.components }
+  if (options.args['--target'] && options.args['--target-content']) {
+    throw new CliError(`You can't set both the 'target' and 'target-content' arguments.`)
+  }
 
-  if (!skipBuild) {
-    await npmRun(dir, 'build')
+  const sceneJson = await getValidSceneJson(options.components, projectRoot)
+  const coords = getBaseCoords(sceneJson)
+  const comps = { components: options.components }
+  const analyticsOpts = {
+    projectHash: await b64HashingFunction(projectRoot),
+    coords
   }
 
-  await build({ args: { '--dir': dir }, ...comps })
+  await options.components.analytics.track('Scene deploy started', analyticsOpts)
 
-  if (options.args['--target'] && options.args['--target-content']) {
-    throw new CliError(`You can't set both the 'target' and 'target-content' arguments.`)
+  if (!skipBuild) {
+    await npmRun(projectRoot, 'build')
   }
 
+  await build({ args: { '--dir': projectRoot }, ...comps })
```

Comment:
this build command isn't the same as `npmRun(projectRoot, 'build')` ?

Reply:
it might be or it might be not, it depends on the content creator preferences. We are also doing these steps in the `start` command. If the user has some custom build step, we want to run it for him + ensure we run our own build also

Reply:
mmm but why ? If i use the command skip build i expect that the `build` command is not going to be called 🤔

Reply:
```
      --skip-build              Skip build before deploy
```

Reply:
we can do:
```ts
if (!skipBuild) {
  await npmRun(projectRoot, 'build')
  await build({ args: { '--dir': projectRoot }, ...comps })
}
```

but I think we should leave both just in case the content creator customizes `npm run build` .

what do you think?

Reply:
if you didn't change your build script then it will be executed twice.
My thoughts about this is that the deploy command should run always npm run build, unless you add the `skip-build` param. This way its easy to understand what is going to do and what is expected.
Then if you update the script for npm run build is up to you. Its like ejecting the react-scripts, we can't do anything there.